zipline/finance/risk/cumulative.py

```diff
@@ -312,10 +312,18 @@ def update(self, dt, algorithm_returns, benchmark_returns, account):
         self.excess_returns[self.latest_dt] = (
             self.algorithm_cumulative_returns[self.latest_dt] -
             self.treasury_period_return)
-        self.metrics.beta[dt] = self.calculate_beta()
+        dt_index = self.metrics.index.searchsorted(dt)
+        algorithm_returns = self.algorithm_returns.values[:dt_index + 1]
```

Comment:
On further review, the dropna is no longer doing right hand trimming, because of the earlier use of `[:dt]` when assigning `self.algorithm_returns`.

Looking at removing some extraneous re-slicing.